docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+ingress policy allows auto-registration of EventTypes, the Broker will create the EventType.
+
+Example: 
+
+Set up a Broker with `autoAdd` enabled.
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: auto-add-demo
+spec:
+  ingressPolicy:
+    autoAdd: true
+```
+
+Now if someone emits a non-registered event (e.g., `dev.knative.foo.bar`)
+into the Broker `auto-add-demo`, an EventType will be created upon the event arrival. 
+Note that the Broker's address is well-known, it will always be
+`<name>-broker.<namespace>.svc.<ending>`. In this case case, it is
+`auto-add-demo-broker.default.svc.cluster.local`.
+
+We can send the event manually. While SSHed into a `Pod` with the Istio sidecar, run:
+
+```shell
+curl -v "http://auto-add-demo-broker.default.svc.cluster.local/" \
+  -X POST \
+  -H "X-B3-Flags: 1" \
+  -H "CE-CloudEventsVersion: 0.1" \
+  -H "CE-EventType: dev.knative.foo.bar" \
+  -H "CE-EventTime: 2018-04-05T03:56:24Z" \
+  -H "CE-EventID: 45a8b444-3213-4758-be3f-540bf93f85ff" \
+  -H "CE-Source: dev.knative.example" \
+  -H 'Content-Type: application/json' \
+  -d '{ "much": "wow" }'
+```
+ 
+The Broker `auto-add-demo` will then create the EventType with type `dev.knative.foo.bar` in the Registry.
+
+## Discoverability Use Case
+
+By adding the spec.* fields of EventType as custom columns in the CRD we can fulfill the discoverability 
+use case: 
+
+*As an Event Consumer I want to be able to discover the different event types that I can consume 
+from the different Brokers, without resorting to any OOB mechanism.*
+
+First, Event Consumers will list the EventTypes registered in the system
+
+`$ kubectl get eventtypes -n default`
+
+```
+NAME                                         TYPE                                    SOURCE                       SCHEMA     BROKER          READY  REASON
+dev.knative.foo.bar-55wcn                    dev.knative.foo.bar                     dev.knative.example                     auto-add-demo   True 
+repofork                                     repo:fork                               my-other-user/my-other-repo             dev             False  BrokerIsNotReady
+repopush                                     repo:push                               my-other-user/my-other-repo  my-schema  default         True 
+dev.knative.source.github.push-34cnb         dev.knative.source.github.push          my-user/my-repo                         default         True 
+dev.knative.source.github.pullrequest-86jhv  dev.knative.source.github.pull_request  my-user/my-repo                         default         True  
+```
+
+Then, they will be able to *easily* create the appropriate Trigger(s)
+
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: my-service-trigger
+  namespace: default
+spec:
+  filter:
+    sourceAndType:
+      type: dev.knative.foo.bar
+      source: dev.knative.example
```

Comment:
This heavily implies that the source in the EventType is identical to the source attribute in the CloudEvent itself, which seems to be explicitly not the case above.

Reply:
Added more explanations to the source field.